### ISSUE TITLE
Customers have a tool to automate remediation of Chef Infra Client version upgrade deprecations within their cookbooks

### ISSUE THREAD

**Issue by jonsmorrow** [chef/chef-workstation#444]
## What problem are we solving?

Upgrading from older versions of Chef Infra Client to Chef Infra Client 15.X takes work. Customers need to know what that work is in order to do it.

Facilitating successful customer transitions from older chef client version to current versions and patterns with minimal customer coding/work.

--

After running this you have a local directory with all of your cookbooks updated to current patterns on a branch ready to check-in.

## Objective and Key Results

**Objective** | **Key Results**
-|-
Consume reports (from WS-E-16) and automatically fix what can be fixed|- Feedback from the field that migration process is easier
- Migration tools are used by 80% of CS teams by 1H 2020

## Deliverables 

(Deliver exec. file. CS team can run via CLI or tray app)

1. Update cookstyle rules to autocorrect things that can be corrected by Chef tools.

**Future:**

Once all the code is fixed and working on Chef 15 we need to help them switch over in production this can be done by setting up pipelines to deploy the new stuff in parallel to the old but turned off. When the customer is ready we can use a cookbook in the nodes existing run list to turn on the hab deployed stuff and turn off/uninstall omnibus chef.

## Evidence

- Foo bar

## Dependencies

- May be some UX need.
- Soft dependency on WS-E-16. We can do all of the cookstyle work without depending on 16, but if there are things that require the report then we'll be dependent.

## Acceptance Criteria

- The tool automatically upgrades cookbooks to chef 15 compatable coding standards for everything except Chef server API (search, data bags, ruby blocks, etc...) usage or other non-automatable fixes.

## Out of Scope

- Tool will not support Chef server API access w/i cookbooks

## Open Questions

1. What is the user journey for WS-E-16 & 17. What's more important to the user and what is less.

## Reference docs

Aha! Link: https://chef.aha.io/epics/WS-E-17

**Comment by tyler-ball** [comment]
Closing this epic because we got as much done on it as we can. Work is continuing in #1016.

There is no way we can "remediate deprecations" in customer cookbooks that does not involve either:
1) automatically fixing cookstyle issues, which can be done in cookstyle already, even for a monorepo
2) involves making cookbook changes and submitting them to the pipeline for users

Users explicitly do not want 2) because they want more control over what, when, and how cookbook changes are sent to their pipeline.
